pkg/cluster/spec.go

```diff
@@ -179,6 +180,11 @@ func (s *Spec) RootVersionsBundle() *VersionsBundle {
 	return s.VersionsBundle(s.Cluster.Spec.KubernetesVersion)
 }
 
+// FirstVersionsBundle returns the VersionsBundle for the Cluster specs first Bundles.Spec.VersionBundles.
+func (s *Spec) FirstVersionsBundle() *releasev1.VersionsBundle {
```

Comment:
Could newSpec.RootVersionBundle() be enough?

Reply:
`spec.RootVersionsBundle` returns a VersionsBundle for the Cluster objects root Kubernetes versions. But I think that we should use the first versions bundle for both the plan and the actual upgrade. This would keep the plan and the upgrades consistent with each other. The CAPI upgrader also uses the VersionsBundle to [write configuration files](https://github.com/aws/eks-anywhere/blob/a5676f32d82ef865455d0a72502c554242167389/pkg/executables/clusterctl.go#L67) and the generated diff to perform the upgrades when calling clusterctl [here](https://github.com/aws/eks-anywhere/blob/a5676f32d82ef865455d0a72502c554242167389/pkg/executables/clusterctl.go#L367), so inconsistencies causes some problems

Reply:
Discussed via slack